azurelinuxagent/ga/exthandlers.py

```diff
@@ -516,18 +557,33 @@ def handle_ext_handlers(self, goal_state_id):
                     if handler_i.get_handler_status() is None:
                         handler_i.set_handler_status(message=depends_on_err_msg, code=-1)
 
-                    handler_i.create_status_file_if_not_exist(extension, status=ExtensionStatusValue.error, code=-1,
-                                                              operation=WALAEventOperation.ExtensionProcessing,
-                                                              message=depends_on_err_msg)
+                    handler_i.create_status_file(extension, status=ExtensionStatusValue.error, code=-1,
+                                                 operation=WALAEventOperation.ExtensionProcessing,
+                                                 message=depends_on_err_msg)
 
                 # For SC extensions, overwrite the HandlerStatus with the relevant message
                 else:
                     handler_i.set_handler_status(message=depends_on_err_msg, code=-1)
 
                 continue
 
+            # Invoke policy engine to determine if extension is allowed. If disallowed, report an error on behalf of
+            # the extension and do not process the extension. Dependent extensions will also be blocked.
+            extension_allowed = policy_engine.should_allow_extension(ext_handler.name)
+            if not extension_allowed:
+                msg = (
+                    "Extension will not be processed: failed to {0} extension '{1}' because it is not specified "
+                    "in the allowlist. To {0}, add the extension to the allowed list in the policy file ('{2}')."
```

Comment:
Here, we use both the terms "allowlist" and "allowed list". Does this make sense? 
Maybe something like "add <ext_name> to the list of allowed extensions in the policy file"?

Reply:
@narrieta how about something like this:
"Extension will not be processed: failed to run extension 'CustomScript' because it is not specified **as an allowed extension**. To run, add the extension to the **list of allowed extensions** in the policy file ('/etc/waagent_policy.json')."

Reply:
Updated the message as stated above